src/loaders/gtdb/parse_computed_genome_attribs.py

```diff
@@ -0,0 +1,223 @@
+from src.loaders.gtdb.gtdb_loader_helper import convert_to_json, init_genome_atrri_doc
+
+# Default result file name for parsed computed GTDB genome attributes data for arango import
+GTDB_COMPUTED_GENOME_ATTR_FILE = "gtdb_computed_genome_attribs.json"
+
+# The following features will be extracted from the CheckM2 result quality_report.tsv file as computed genome attributes
+SELECTED_CHECKM2_FEATURES = {'Completeness', 'Contamination'}
+
+# The following features will be extracted from the GTDB-TK summary file
+# ('gtdbtk.ar53.summary.tsv' or 'gtdbtk.bac120.summary.tsv') as computed genome attributes
+SELECTED_GTDBTK_SUMMARY_FEATURES = {'classification', 'fastani_taxonomy', 'closest_placement_taxonomy',
+                                    'pplacer_taxonomy'}
+
+NOTIFICATION = ' This argument should be consistent with the one used by the gtdb_genome_attribs_loader.py script.'
+
+
+def _locate_dir(root_dir, kbase_collection, load_ver, check_exists=False, tool=''):
+    result_dir = os.path.join(root_dir, loader_common_names.COLLECTION_DATA_DIR, kbase_collection, load_ver, tool)
+
+    if check_exists and not (os.path.exists(result_dir) and os.path.isdir(result_dir)):
+        raise ValueError(f"Result directory for computed genome attributes of "
+                         f"Kbase Collection: {kbase_collection} and Load Version: {load_ver} could not be found.")
+
+    return result_dir
+
+
+def _read_tsv_as_df(file_path, features, genome_id_col=None):
+    # Retrieve the desired fields from a TSV file and return the data in a dataframe
+
+    selected_cols = copy.deepcopy(features)
+    if genome_id_col:
+        selected_cols.add(genome_id_col)  # genome_id col is used to parse genome id
+    df = pd.read_csv(file_path, sep='\t', keep_default_na=False, usecols=selected_cols)
+
+    return df
+
+
+def _create_doc(row, kbase_collection, load_version, genome_id, selected_features, pre_fix='KBase'):
+    # Select specific columns and prepare them for import into Arango
+
+    # NOTE: The selected column names will have a prefix (the tool name) added to them to distinguish them from
+    #       the original metadata column names.
+
+    doc = init_genome_atrri_doc(kbase_collection, load_version, genome_id)
+
+    # distinguish the selected fields from the original metadata by adding a common prefix to their names
+    doc.update(row[list(selected_features)].rename(lambda x: pre_fix + '_' + x).to_dict())
+
+    return doc
+
+
+def _gtdbtk_row_to_doc(row, kbase_collection, load_version):
+    # Transforms row from GTDBTK summary file 'gtdbtk.ar53.summary.tsv' or 'gtdbtk.bac120.summary.tsv'
+    # into ArangoDB collection document
+
+    # parse genome id from the "user_genome" field.
+    # the "user_genome" field of GTDBTK summary is the name of input file
+    # (e.g. GCA_019057475.1_ASM1905747v1_genomic.fna.gz, GCF_000970085.1_ASM97008v1_genomic.fna.gz).
+    genome_id = '_'.join(row.user_genome.split('_')[:2])
```

Comment:
Again I'm wondering how this is going to work for non-gtdb collections

Reply:
I was thinking it would be the same as long as the source files for tools are coming from NCBI. It seems like this might not be true for PMI. Without PMI source data, I really don't know what the output will look like. I don't know what the parsing logic should be for non-NCBI source files.

Reply:
Covered in comment below this review

Reply:
:+1: